lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -179,6 +179,13 @@ LogicalResult calyx::verifyCell(Operation *op) {
 
 LogicalResult calyx::verifyControlLikeOp(Operation *op) {
   auto parent = op->getParentOp();
+
+  if (isa<calyx::EnableOp>(op) &&
+      !isa<calyx::CalyxDialect>(parent->getDialect())) {
```

Comment:
This should be restricted to fsm::FSMDialect, no?

Reply:
I think "anything goes" might be preferable to taking a dependency on the FSMDialect, and tightly coupling the two.

Reply:
That's exactly what we're doing here: coupling the FSMDialect and Calyx enable operations. Why open it to other dialects as well? I think either is fine, just curious mostly why this choice was made.

Reply:
IOW, if I was reading this code without having seen the FSM discussion 5 minutes ago, I would be confused why this exception exists.

Reply:
My point was that making this exception explicit in terms of the FSM dialect would mean a library-level dependency from the Calyx dialect to the FSM dialect. This may not be a problem, if we are committed to co-designing the two dialects. But in general, dialect dependencies of this manner need to be thought through, so we don't paint ourselves into a corner with cyclic dependencies.

The way this is implemented here takes a different angle: it's saying the Calyx dialect no longer has a hard-line stance that Calyx is a closed world that can't be composed with other dialects, but it does so without naming those dialects, avoiding library-level dependencies.

I think this is actually the main sticking point in @mortbopet's progressive lowering to the FSM dialect. Are we ready to relax Calyx's closed world view? If so, should we relax it to also permit the FSM dialect explicitly, and commit to maintaining that dependency, or should it be open-ended? If not, then perhaps the lowering to FSM should be "whole hog" and not stop in this intermediate state where Calyx enables are embedded in the FSM.

I personally think there is a strong argument for keeping Calyx "closed world", since it is a dialect that is meant for interchange between two compiler infrastructures. This is why I was lobbying for the FSM lowering to be "whole hog", rather than permitting this intermediate state and requiring the change in this PR.